## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 388787`

Logs: https://github.com/GaetanLepage/nixpkgs-review-gha/actions/runs/16020149270


---
### `x86_64-linux` (sandbox = true)
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>chatty</li>
  </ul>
</details>
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>calls</li>
    <li>calls.devdoc</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 193 packages built:</summary>
  <ul>
    <li>CuboCore.coreaction</li>
    <li>CuboCore.corearchiver</li>
    <li>CuboCore.corefm</li>
    <li>CuboCore.coregarage</li>
    <li>CuboCore.corehunt</li>
    <li>CuboCore.coreimage</li>
    <li>CuboCore.coreinfo</li>
    <li>CuboCore.corekeyboard</li>
    <li>CuboCore.corepad</li>
    <li>CuboCore.corepaint</li>
    <li>CuboCore.corepdf</li>
    <li>CuboCore.corepins</li>
    <li>CuboCore.corerenamer</li>
    <li>CuboCore.coreshot</li>
    <li>CuboCore.corestats</li>
    <li>CuboCore.coreterminal</li>
    <li>CuboCore.coretime</li>
    <li>CuboCore.coretoppings</li>
    <li>CuboCore.coreuniverse</li>
    <li>CuboCore.libcsys</li>
    <li>adapta-gtk-theme</li>
    <li>almanah</li>
    <li>apacheHttpdPackages.mod_tile (apacheHttpdPackages_2_4.mod_tile)</li>
    <li>astal.cava</li>
    <li>astal.mpris</li>
    <li>ayatana-indicator-datetime</li>
    <li>blivet-gui</li>
    <li>blivet-gui.dist</li>
    <li>bubblemail</li>
    <li>bubblemail.dist</li>
    <li>budgie-control-center</li>
    <li>budgie-control-center.debug</li>
    <li>cantata</li>
    <li>cava</li>
    <li>cavalcade</li>
    <li>cavalcade.dist</li>
    <li>cavalier</li>
    <li>cheese</li>
    <li>cheese.devdoc</li>
    <li>cheese.man</li>
    <li>cinnamon-common</li>
    <li>cinnamon-gsettings-overrides</li>
    <li>cinnamon-screensaver</li>
    <li>clementine</li>
    <li>deepin.dde-control-center</li>
    <li>deepin.dde-control-center.dev</li>
    <li>deepin.dde-device-formatter</li>
    <li>deepin.dde-file-manager</li>
    <li>deepin.dde-gsettings-schemas</li>
    <li>deepin.dde-network-core</li>
    <li>deepin.dde-session-shell</li>
    <li>deepin.dde-session-shell.dev</li>
    <li>deepin.deepin-anything</li>
    <li>deepin.deepin-compressor</li>
    <li>deepin.deepin-pw-check</li>
    <li>deepin.udisks2-qt5</li>
    <li>deepin.util-dfm</li>
    <li>endeavour</li>
    <li>enlightenment.enlightenment</li>
    <li>eos-installer</li>
    <li>evolution</li>
    <li>evolution-data-server</li>
    <li>evolution-data-server-gtk4</li>
    <li>evolution-data-server-gtk4.dev</li>
    <li>evolution-data-server.dev</li>
    <li>evolution-ews</li>
    <li>evolutionWithPlugins</li>
    <li>exaile</li>
    <li>folks</li>
    <li>folks.dev</li>
    <li>folks.devdoc</li>
    <li>geary</li>
    <li>gitg</li>
    <li>gnome-applets</li>
    <li>gnome-browser-connector</li>
    <li>gnome-calendar</li>
    <li>gnome-contacts</li>
    <li>gnome-control-center</li>
    <li>gnome-control-center.debug</li>
    <li>gnome-disk-utility</li>
    <li>gnome-flashback</li>
    <li>gnome-multi-writer</li>
    <li>gnome-music</li>
    <li>gnome-notes</li>
    <li>gnome-online-accounts</li>
    <li>gnome-online-accounts-gtk</li>
    <li>gnome-online-accounts.debug</li>
    <li>gnome-online-accounts.dev</li>
    <li>gnome-online-accounts.devdoc</li>
    <li>gnome-online-accounts.man</li>
    <li>gnome-panel</li>
    <li>gnome-panel-with-modules</li>
    <li>gnome-panel.dev</li>
    <li>gnome-panel.man</li>
    <li>gnome-photos</li>
    <li>gnome-photos.installedTests</li>
    <li>gnome-recipes</li>
    <li>gnome-session</li>
    <li>gnome-session.debug</li>
    <li>gnome-session.sessions</li>
    <li>gnome-shell</li>
    <li>gnome-shell.debug</li>
    <li>gnome-shell.devdoc</li>
    <li>gnome-tweaks</li>
    <li>gnome.gvfs</li>
    <li>gnome.gvfs.debug</li>
    <li>gnome.nixos-gsettings-overrides</li>
    <li>gnomeExtensions.easyScreenCast</li>
    <li>gnomeExtensions.gsconnect</li>
    <li>gnomeExtensions.gsconnect.installedTests</li>
    <li>grilo-plugins</li>
    <li>gvfs</li>
    <li>gvfs.debug</li>
    <li>hardinfo2</li>
    <li>hyprpanel</li>
    <li>iniparser</li>
    <li>libblockdev</li>
    <li>libblockdev.dev</li>
    <li>libblockdev.devdoc</li>
    <li>libblockdev.python</li>
    <li>libcava</li>
    <li>libgdata</li>
    <li>libgdata.dev</li>
    <li>libgdata.installedTests</li>
    <li>libmsgraph</li>
    <li>libmsgraph.dev</li>
    <li>libmsgraph.devdoc</li>
    <li>libndctl</li>
    <li>libndctl.dev</li>
    <li>libndctl.man</li>
    <li>lomiri.lomiri-music-app</li>
    <li>lomiri.mediascanner2</li>
    <li>lomiri.mediascanner2.dev</li>
    <li>marble-shell-theme</li>
    <li>mate.mate-control-center</li>
    <li>mate.mate-panel-with-applets</li>
    <li>mate.mate-settings-daemon-wrapped</li>
    <li>mate.mate-utils</li>
    <li>matrix-gtk-theme</li>
    <li>mediawriter</li>
    <li>mojave-gtk-theme</li>
    <li>monitor</li>
    <li>mpd-notification</li>
    <li>nemo</li>
    <li>nemo-fileroller</li>
    <li>nemo-preview</li>
    <li>nemo-python</li>
    <li>nemo-seahorse</li>
    <li>nemo-with-extensions</li>
    <li>nemo.dev</li>
    <li>netatalk</li>
    <li>ns-usbloader</li>
    <li>pantheon.elementary-calendar</li>
    <li>pantheon.elementary-greeter</li>
    <li>pantheon.elementary-mail</li>
    <li>pantheon.elementary-session-settings</li>
    <li>pantheon.elementary-tasks</li>
    <li>pantheon.switchboard-plug-about</li>
    <li>pantheon.switchboard-plug-onlineaccounts</li>
    <li>pantheon.switchboard-with-plugs</li>
    <li>pantheon.wingpanel-applications-menu</li>
    <li>pantheon.wingpanel-indicator-datetime</li>
    <li>pantheon.wingpanel-with-indicators</li>
    <li>phosh</li>
    <li>phosh-mobile-settings</li>
    <li>planify</li>
    <li>plasticity</li>
    <li>python312Packages.blivet</li>
    <li>python312Packages.blivet.dist</li>
    <li>python313Packages.blivet</li>
    <li>python313Packages.blivet.dist</li>
    <li>rapid-photo-downloader</li>
    <li>rapid-photo-downloader.dist</li>
    <li>spaceFM</li>
    <li>swaynotificationcenter</li>
    <li>swaysettings</li>
    <li>themechanger</li>
    <li>tokyonight-gtk-theme</li>
    <li>totem</li>
    <li>udiskie</li>
    <li>udiskie.dist</li>
    <li>udisks (udisks2)</li>
    <li>udisks.dev (udisks2.dev)</li>
    <li>udisks.devdoc (udisks2.devdoc)</li>
    <li>udisks.man (udisks2.man)</li>
    <li>unity-test</li>
    <li>unity-test.dev</li>
    <li>usbimager</li>
    <li>usermount</li>
    <li>valent</li>
    <li>vifm-full</li>
    <li>vimix-gtk-themes</li>
    <li>waybar</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>calls</summary>
<pre>[460/475] Linking target plugins/provider/tests/sip
[461/475] Linking target plugins/provider/tests/srtp
[462/475] Linking target plugins/provider/tests/sdp-crypto
[463/475] Linking target tests/emergency-call-types
[464/475] Linking target tests/util
[465/475] Linking target tests/services/calls-dbus-server
[466/475] Linking target tests/manager
[467/475] Linking target tests/plugins
[468/475] Linking target tests/ui-call
[469/475] Linking target tests/settings
[470/475] Linking target tests/dbus
[471/475] Linking target tests/ringer
[472/475] Linking target tests/contacts
[473/475] Linking target doc/libcalls-doc.so
[474/475] Generating doc/man-gnome-calls with a custom command (wrapped by meson to capture output)
[475/475] Linking target tests/application
buildPhase completed in 4 minutes 28 seconds
Running phase: checkPhase
@nix { "action": "setPhase", "phase": "checkPhase" }
bwrap: setting up uid map: Permission denied</pre>
</details>
</details>


---
### `aarch64-linux` (sandbox = true)
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>chatty</li>
  </ul>
</details>
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>calls</li>
    <li>calls.devdoc</li>
    <li>deepin.dde-gsettings-schemas</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 184 packages built:</summary>
  <ul>
    <li>CuboCore.coreaction</li>
    <li>CuboCore.corearchiver</li>
    <li>CuboCore.corefm</li>
    <li>CuboCore.coregarage</li>
    <li>CuboCore.corehunt</li>
    <li>CuboCore.coreimage</li>
    <li>CuboCore.coreinfo</li>
    <li>CuboCore.corekeyboard</li>
    <li>CuboCore.corepad</li>
    <li>CuboCore.corepaint</li>
    <li>CuboCore.corepdf</li>
    <li>CuboCore.corepins</li>
    <li>CuboCore.corerenamer</li>
    <li>CuboCore.coreshot</li>
    <li>CuboCore.corestats</li>
    <li>CuboCore.coreterminal</li>
    <li>CuboCore.coretime</li>
    <li>CuboCore.coretoppings</li>
    <li>CuboCore.coreuniverse</li>
    <li>CuboCore.libcsys</li>
    <li>adapta-gtk-theme</li>
    <li>almanah</li>
    <li>apacheHttpdPackages.mod_tile (apacheHttpdPackages_2_4.mod_tile)</li>
    <li>astal.cava</li>
    <li>astal.mpris</li>
    <li>ayatana-indicator-datetime</li>
    <li>bubblemail</li>
    <li>bubblemail.dist</li>
    <li>budgie-control-center</li>
    <li>budgie-control-center.debug</li>
    <li>cantata</li>
    <li>cava</li>
    <li>cavalcade</li>
    <li>cavalcade.dist</li>
    <li>cavalier</li>
    <li>cheese</li>
    <li>cheese.devdoc</li>
    <li>cheese.man</li>
    <li>cinnamon-common</li>
    <li>cinnamon-gsettings-overrides</li>
    <li>cinnamon-screensaver</li>
    <li>clementine</li>
    <li>deepin.dde-control-center</li>
    <li>deepin.dde-control-center.dev</li>
    <li>deepin.dde-device-formatter</li>
    <li>deepin.dde-file-manager</li>
    <li>deepin.dde-network-core</li>
    <li>deepin.dde-session-shell</li>
    <li>deepin.dde-session-shell.dev</li>
    <li>deepin.deepin-anything</li>
    <li>deepin.deepin-compressor</li>
    <li>deepin.deepin-pw-check</li>
    <li>deepin.udisks2-qt5</li>
    <li>deepin.util-dfm</li>
    <li>endeavour</li>
    <li>enlightenment.enlightenment</li>
    <li>eos-installer</li>
    <li>evolution</li>
    <li>evolution-data-server</li>
    <li>evolution-data-server-gtk4</li>
    <li>evolution-data-server-gtk4.dev</li>
    <li>evolution-data-server.dev</li>
    <li>evolution-ews</li>
    <li>evolutionWithPlugins</li>
    <li>exaile</li>
    <li>folks</li>
    <li>folks.dev</li>
    <li>folks.devdoc</li>
    <li>geary</li>
    <li>gitg</li>
    <li>gnome-applets</li>
    <li>gnome-browser-connector</li>
    <li>gnome-calendar</li>
    <li>gnome-contacts</li>
    <li>gnome-control-center</li>
    <li>gnome-control-center.debug</li>
    <li>gnome-disk-utility</li>
    <li>gnome-flashback</li>
    <li>gnome-multi-writer</li>
    <li>gnome-music</li>
    <li>gnome-notes</li>
    <li>gnome-online-accounts</li>
    <li>gnome-online-accounts-gtk</li>
    <li>gnome-online-accounts.debug</li>
    <li>gnome-online-accounts.dev</li>
    <li>gnome-online-accounts.devdoc</li>
    <li>gnome-online-accounts.man</li>
    <li>gnome-panel</li>
    <li>gnome-panel-with-modules</li>
    <li>gnome-panel.dev</li>
    <li>gnome-panel.man</li>
    <li>gnome-photos</li>
    <li>gnome-photos.installedTests</li>
    <li>gnome-recipes</li>
    <li>gnome-session</li>
    <li>gnome-session.debug</li>
    <li>gnome-session.sessions</li>
    <li>gnome-shell</li>
    <li>gnome-shell.debug</li>
    <li>gnome-shell.devdoc</li>
    <li>gnome-tweaks</li>
    <li>gnome.gvfs</li>
    <li>gnome.gvfs.debug</li>
    <li>gnome.nixos-gsettings-overrides</li>
    <li>gnomeExtensions.easyScreenCast</li>
    <li>gnomeExtensions.gsconnect</li>
    <li>gnomeExtensions.gsconnect.installedTests</li>
    <li>grilo-plugins</li>
    <li>gvfs</li>
    <li>gvfs.debug</li>
    <li>hardinfo2</li>
    <li>hyprpanel</li>
    <li>iniparser</li>
    <li>libblockdev</li>
    <li>libblockdev.dev</li>
    <li>libblockdev.devdoc</li>
    <li>libblockdev.python</li>
    <li>libcava</li>
    <li>libgdata</li>
    <li>libgdata.dev</li>
    <li>libgdata.installedTests</li>
    <li>libmsgraph</li>
    <li>libmsgraph.dev</li>
    <li>libmsgraph.devdoc</li>
    <li>libndctl</li>
    <li>libndctl.dev</li>
    <li>libndctl.man</li>
    <li>lomiri.lomiri-music-app</li>
    <li>lomiri.mediascanner2</li>
    <li>lomiri.mediascanner2.dev</li>
    <li>marble-shell-theme</li>
    <li>mate.mate-control-center</li>
    <li>mate.mate-panel-with-applets</li>
    <li>mate.mate-settings-daemon-wrapped</li>
    <li>mate.mate-utils</li>
    <li>matrix-gtk-theme</li>
    <li>mediawriter</li>
    <li>mojave-gtk-theme</li>
    <li>monitor</li>
    <li>mpd-notification</li>
    <li>nemo</li>
    <li>nemo-fileroller</li>
    <li>nemo-preview</li>
    <li>nemo-python</li>
    <li>nemo-seahorse</li>
    <li>nemo-with-extensions</li>
    <li>nemo.dev</li>
    <li>netatalk</li>
    <li>ns-usbloader</li>
    <li>pantheon.elementary-calendar</li>
    <li>pantheon.elementary-greeter</li>
    <li>pantheon.elementary-mail</li>
    <li>pantheon.elementary-session-settings</li>
    <li>pantheon.elementary-tasks</li>
    <li>pantheon.switchboard-plug-about</li>
    <li>pantheon.switchboard-plug-onlineaccounts</li>
    <li>pantheon.switchboard-with-plugs</li>
    <li>pantheon.wingpanel-applications-menu</li>
    <li>pantheon.wingpanel-indicator-datetime</li>
    <li>pantheon.wingpanel-with-indicators</li>
    <li>phosh</li>
    <li>phosh-mobile-settings</li>
    <li>planify</li>
    <li>rapid-photo-downloader</li>
    <li>rapid-photo-downloader.dist</li>
    <li>spaceFM</li>
    <li>swaynotificationcenter</li>
    <li>swaysettings</li>
    <li>themechanger</li>
    <li>tokyonight-gtk-theme</li>
    <li>totem</li>
    <li>udiskie</li>
    <li>udiskie.dist</li>
    <li>udisks (udisks2)</li>
    <li>udisks.dev (udisks2.dev)</li>
    <li>udisks.devdoc (udisks2.devdoc)</li>
    <li>udisks.man (udisks2.man)</li>
    <li>unity-test</li>
    <li>unity-test.dev</li>
    <li>usermount</li>
    <li>valent</li>
    <li>vifm-full</li>
    <li>vimix-gtk-themes</li>
    <li>waybar</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>calls</summary>
<pre>[460/475] Linking target plugins/provider/tests/media
[461/475] Linking target plugins/provider/tests/srtp
[462/475] Linking target tests/emergency-call-types
[463/475] Linking target plugins/provider/tests/sdp-crypto
[464/475] Linking target tests/util
[465/475] Linking target tests/manager
[466/475] Linking target tests/services/calls-dbus-server
[467/475] Linking target tests/plugins
[468/475] Linking target tests/settings
[469/475] Linking target tests/ui-call
[470/475] Linking target tests/dbus
[471/475] Linking target tests/contacts
[472/475] Linking target tests/ringer
[473/475] Linking target doc/libcalls-doc.so
[474/475] Generating doc/man-gnome-calls with a custom command (wrapped by meson to capture output)
[475/475] Linking target tests/application
buildPhase completed in 3 minutes 21 seconds
Running phase: checkPhase
@nix { "action": "setPhase", "phase": "checkPhase" }
bwrap: setting up uid map: Permission denied</pre>
</details>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 7 packages failed to build:</summary>
  <ul>
    <li>cava</li>
    <li>cavalcade</li>
    <li>cavalcade.dist</li>
    <li>iniparser</li>
    <li>libcava</li>
    <li>unity-test</li>
    <li>unity-test.dev</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>unity-test</summary>
<pre> 1603 |         for (i = 0; expected[i] || actual[i]; i++)
      |                     ^~~~~~~~
../src/unity.c:1603:36: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
 1603 |         for (i = 0; expected[i] || actual[i]; i++)
      |                                    ^~~~~~
../src/unity.c:1605:17: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
 1605 |             if (expected[i] != actual[i])
      |                 ^~~~~~~~
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
In file included from tests/test_unity_arrays.c:10:
tests/self_assessment_utils.h:89:5: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
   89 |     putcharSpyBuffer[indexSpyBuffer] = '\0';
      |     ^~~~~~~~~~~~~~~~
tests/self_assessment_utils.h:102:13: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
  102 |             putcharSpyBuffer[indexSpyBuffer++] = (char)c;
      |             ^~~~~~~~~~~~~~~~
2 errors generated.
make: *** [Makefile:108: test] Error 1
make: Leaving directory '/private/tmp/nix-build-unity-test-2.6.1.drv-0/source/test'</pre>
</details>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 7 packages failed to build:</summary>
  <ul>
    <li>cava</li>
    <li>cavalcade</li>
    <li>cavalcade.dist</li>
    <li>iniparser</li>
    <li>libcava</li>
    <li>unity-test</li>
    <li>unity-test.dev</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>unity-test</summary>
<pre> 1603 |         for (i = 0; expected[i] || actual[i]; i++)
      |                     ^~~~~~~~
../src/unity.c:1603:36: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
 1603 |         for (i = 0; expected[i] || actual[i]; i++)
      |                                    ^~~~~~
../src/unity.c:1605:17: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
 1605 |             if (expected[i] != actual[i])
      |                 ^~~~~~~~
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
In file included from tests/test_unity_arrays.c:10:
tests/self_assessment_utils.h:89:5: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
   89 |     putcharSpyBuffer[indexSpyBuffer] = '\0';
      |     ^~~~~~~~~~~~~~~~
tests/self_assessment_utils.h:102:13: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
  102 |             putcharSpyBuffer[indexSpyBuffer++] = (char)c;
      |             ^~~~~~~~~~~~~~~~
2 errors generated.
make: *** [Makefile:108: test] Error 1
make: Leaving directory '/private/tmp/nix-build-unity-test-2.6.1.drv-0/source/test'</pre>
</details>
</details>

--- a/pkgs/by-name/in/iniparser/package.nix
+++ b/pkgs/by-name/in/iniparser/package.nix
@@ -10,6 +10,8 @@
   ruby,
   validatePkgConfig,
   testers,
+  unity-test,
+  ctestCheckHook,
 }:
 
 stdenv.mkDerivation (finalAttrs: {
@@ -19,56 +21,57 @@ stdenv.mkDerivation (finalAttrs: {
   src = fetchFromGitLab {
     owner = "iniparser";
     repo = "iniparser";
-    rev = "v${finalAttrs.version}";
+    tag = "v${finalAttrs.version}";
     hash = "sha256-z10S9ODLprd7CbL5Ecgh7H4eOwTetYwFXiWBUm6fIr4=";
   };
 
-  patches = lib.optionals finalAttrs.finalPackage.doCheck [
-    (replaceVars ./remove-fetchcontent-usage.patch {
-      # Do not let cmake's fetchContent download unity
-      unitySrc = symlinkJoin {
-        paths = [
-          (fetchFromGitHub {
-            owner = "throwtheswitch";
-            repo = "unity";
-            rev = "v2.6.0";
-            hash = "sha256-SCcUGNN/UJlu3ALJiZ9bQKxYRZey3cm9QG+NOehp6Ow=";
-          })
-        ];
-        postBuild = ''
-          ln -s ${finalAttrs.src}/test/unity_config.h $out/src/unity_config.h
-        '';
-      };
-    })
-  ];
+  patches = lib.optional finalAttrs.doCheck (
+    # 1. Do not fetch the Unity GitHub repository
+    # 2. Lookup the Unity pkgconfig file
+    # 3. Get the generate_test_runner.rb file from the Unity share directory
+    replaceVars ./remove-fetchcontent-usage.patch {
+      # Get the test generator
+      UNITY-GENERATE-TEST-RUNNER = "${unity-test}/share/generate_test_runner.rb";
+    }
+  );
 
   nativeBuildInputs = [
     cmake
     doxygen
     validatePkgConfig
-  ] ++ lib.optionals finalAttrs.finalPackage.doCheck [ ruby ];
+  ];
 
-  cmakeFlags = [ "-DBUILD_TESTING=${if finalAttrs.finalPackage.doCheck then "ON" else "OFF"}" ];
-
-  doCheck = false;
+  cmakeFlags = [
+    (lib.cmakeBool "BUILD_TESTING" finalAttrs.doCheck)
+  ];
+  doCheck = true;
+  nativeCheckInputs = [
+    ruby
+    ctestCheckHook
+  ];
+  checkInputs = [
+    (
+      (unity-test.override {
+        supportDouble = true;
+      }).overrideAttrs
+      {
+        doCheck = false;
+      }
+    )
+  ];
 
   postFixup = ''
     ln -sv $out/include/iniparser/*.h $out/include/
   '';
 
-  passthru.tests = {
-    pkg-config = testers.testMetaPkgConfig finalAttrs.finalPackage;
-    iniparser-with-tests = finalAttrs.overrideAttrs (_: {
-      doCheck = true;
-    });
-  };
+  passthru.tests.pkg-config = testers.testMetaPkgConfig finalAttrs.finalPackage;
 
-  meta = with lib; {
+  meta = {
     homepage = "https://gitlab.com/iniparser/iniparser";
     description = "Free standalone ini file parsing library";
     changelog = "https://gitlab.com/iniparser/iniparser/-/releases/v${finalAttrs.version}";
-    license = licenses.mit;
-    platforms = platforms.unix;
+    license = lib.licenses.mit;
+    platforms = lib.platforms.unix;
     pkgConfigModules = [ "iniparser" ];
     maintainers = [ ];
   };

--- a/pkgs/by-name/in/iniparser/remove-fetchcontent-usage.patch
+++ b/pkgs/by-name/in/iniparser/remove-fetchcontent-usage.patch
@@ -1,17 +1,52 @@
 diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
-index b28d151..33a6bcf 100644
+index 0735d27..32c5cdb 100644
 --- a/test/CMakeLists.txt
 +++ b/test/CMakeLists.txt
-@@ -28,10 +28,8 @@ set(FETCHCONTENT_QUIET OFF)
+@@ -26,16 +26,8 @@ endif()
  
- FetchContent_Declare(
-   unity
+ set(FETCHCONTENT_QUIET OFF)
+ 
+-FetchContent_Declare(
+-  unity
 -  GIT_REPOSITORY "https://github.com/throwtheswitch/unity.git"
 -  GIT_PROGRESS TRUE
 -  PATCH_COMMAND ${CMAKE_COMMAND} -E copy
 -                ${CMAKE_CURRENT_LIST_DIR}/unity_config.h ./src/)
-+  SOURCE_DIR @unitySrc@
-+)
+-
+-FetchContent_MakeAvailable(unity)
+-target_compile_definitions(unity PUBLIC UNITY_INCLUDE_CONFIG_H
+-                                        UNITY_USE_COMMAND_LINE_ARGS)
++find_package(PkgConfig REQUIRED)
++pkg_check_modules(UNITY REQUIRED unity)
  
- FetchContent_MakeAvailable(unity)
- target_compile_definitions(unity PUBLIC UNITY_INCLUDE_CONFIG_H)
+ function(create_test_runner)
+   set(options)
+@@ -52,7 +44,7 @@ function(create_test_runner)
+   add_custom_command(
+     OUTPUT test_${TEST_RUNNER_NAME}_runner.c
+     COMMAND
+-      ${RUBY_EXECUTABLE} ${unity_SOURCE_DIR}/auto/generate_test_runner.rb
++      @UNITY-GENERATE-TEST-RUNNER@
+       ${CMAKE_CURRENT_SOURCE_DIR}/test_${TEST_RUNNER_NAME}.c
+       test_${TEST_RUNNER_NAME}_runner.c ${CMAKE_CURRENT_LIST_DIR}/unity-config.yml
+     DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/test_${TEST_RUNNER_NAME}.c
+@@ -62,10 +54,18 @@ function(create_test_runner)
+                                           test_${TEST_RUNNER_NAME}_runner.c)
+   foreach(TARGET_TYPE ${TARGET_TYPES})
+     # if BUILD_STATIC_LIBS=ON shared takes precedence
++    target_include_directories(
++      test_${TEST_RUNNER_NAME}
++      PUBLIC
++      ${UNITY_INCLUDE_DIRS})
++    target_compile_options(
++      test_${TEST_RUNNER_NAME}
++      PUBLIC
++      ${UNITY_CFLAGS_OTHER})
+     target_link_libraries(
+       test_${TEST_RUNNER_NAME}
+       ${PROJECT_NAME}-${TARGET_TYPE}
+-      unity)
++      ${UNITY_LIBRARIES})
+   endforeach()
+ endfunction()
+ 

--- a/pkgs/by-name/un/unity-test/meson.patch
+++ b/pkgs/by-name/un/unity-test/meson.patch
@@ -1,0 +1,18 @@
+diff --git a/meson.build b/meson.build
+index 6585129..9489aef 100644
+--- a/meson.build
++++ b/meson.build
+@@ -64,10 +64,10 @@ unity_dep = declare_dependency(
+ if not meson.is_subproject()
+   pkg = import('pkgconfig')
+   pkg.generate(
+-    name: meson.project_name(),
++    unity_lib,
+     version: meson.project_version(),
+-    libraries: [ unity_lib ],
+-    description: 'C Unit testing framework.'
++    subdirs: 'unity',
++    extra_cflags: unity_args,
+   )
+ endif
+ 

--- a/pkgs/by-name/un/unity-test/package.nix
+++ b/pkgs/by-name/un/unity-test/package.nix
@@ -2,9 +2,33 @@
   lib,
   stdenv,
   fetchFromGitHub,
-  cmake,
-}:
+  fetchpatch2,
+  meson,
+  ninja,
+  ruby,
+  python3Minimal,
+  nix-update-script,
+  testers,
+  iniparser,
+  validatePkgConfig,
+  # Adds test groups and extra CLI flags.
+  buildFixture ? false,
+  # Adds the ablilty to track malloc and free calls.
+  # Note that if fixtures are enabled, this option is ignored
+  # and will always be enabled.
+  buildMemory ? buildFixture,
+  # Adds double precision floating point assertions
+  supportDouble ? false,
 
+}:
+let
+  # On newer versions of Clang, Weverything is too much of everything.
+  ignoredErrors = [
+    "-Wno-unsafe-buffer-usage"
+    "-Wno-reserved-identifier"
+    "-Wno-extra-semi-stmt"
+  ];
+in
 stdenv.mkDerivation (finalAttrs: {
   pname = "unity-test";
   version = "2.6.1";
@@ -16,13 +40,85 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-g0ubq7RxGQmL1R6vz9RIGJpVWYsgrZhsTWSrL1ySEug=";
   };
 
-  nativeBuildInputs = [ cmake ];
+  patches = [
+    # The meson file does not have the subdir set correctly
+    (fetchpatch2 {
+      url = "https://patch-diff.githubusercontent.com/raw/ThrowTheSwitch/Unity/pull/771.patch";
+      hash = "sha256-r8ldVb7WrzVwTC2CtGul9Jk4Rzt+6ejk+paYAfFlR5M=";
+    })
+    # Fix up the shebangs in the auto directory as not all are correct
+    (fetchpatch2 {
+      url = "https://patch-diff.githubusercontent.com/raw/ThrowTheSwitch/Unity/pull/790.patch";
+      hash = "sha256-K+OxMe/ZMXPPjZXjGhgc5ULLN7plBwL0hV5gwmgA3FM=";
+    })
+  ];
+
+  postPatch = ''
+    patchShebangs --build auto
+  '';
+
+  outputs = [
+    "out"
+    "dev"
+  ];
+
+  strictDeps = true;
+  nativeBuildInputs = [
+    meson
+    ninja
+    python3Minimal
+    validatePkgConfig
+  ];
+
+  # For the helper shebangs
+  buildInputs = [
+    python3Minimal
+    ruby
+  ];
+
+  mesonFlags = [
+    (lib.mesonBool "extension_memory" buildMemory)
+    (lib.mesonBool "extension_fixture" buildFixture)
+    (lib.mesonBool "support_double" supportDouble)
+  ];
+
   doCheck = true;
+
+  checkPhase = ''
+    runHook preCheck
+
+    make -C../test -j $NIX_BUILD_CORES ${lib.optionalString stdenv.cc.isClang "CC=clang"} E="-Weverything ${lib.escapeShellArgs ignoredErrors}" test
+
+    runHook postCheck
+  '';
+
+  # Various helpers
+  postInstall = ''
+    mkdir -p "$out/share"
+    install -Dm755 ../auto/* -t "$out/share/"
+  '';
+
+  passthru = {
+    updateScript = nix-update-script { };
+    tests = {
+      inherit iniparser;
+      pkg-config = testers.hasPkgConfigModules {
+        package = finalAttrs.finalPackage;
+        versionCheck = true;
+      };
+    };
+  };
 
   meta = {
     description = "Unity Unit Testing Framework";
     homepage = "https://www.throwtheswitch.org/unity";
+    changelog = "https://github.com/ThrowTheSwitch/Unity/releases/tag/v${finalAttrs.version}";
     license = lib.licenses.mit;
-    maintainers = [ lib.maintainers.i01011001 ];
+    platforms = lib.platforms.all;
+    pkgConfigModules = [ "unity" ];
+    maintainers = with lib.maintainers; [
+      i01011001
+      RossSmyth
+    ];
   };
 })

